src/ParallelAlgorithms/Amr/Actions/AdjustDomain.hpp

```diff
@@ -101,44 +98,62 @@ struct AdjustDomain {
       }
 
     } else {
+      const auto old_mesh_and_element =
+          std::make_pair(db::get<::domain::Tags::Mesh<volume_dim>>(box),
+                         db::get<::domain::Tags::Element<volume_dim>>(box));
+      const auto& old_mesh = old_mesh_and_element.first;
+
+      // Determine new neighbors and update the Element
+      {  // avoid shadowing when mutating flags below
+        const auto& amr_flags_of_neighbors =
+            db::get<amr::Tags::NeighborFlags<volume_dim>>(box);
+        db::mutate<::domain::Tags::Element<volume_dim>>(
+            [&element_id, &amr_flags_of_neighbors](
+                const gsl::not_null<Element<volume_dim>*> element) {
+              auto new_neighbors = element->neighbors();
+              for (auto& [direction, neighbors] : new_neighbors) {
+                neighbors.set_ids_to(amr::new_neighbor_ids(
+                    element_id, direction, neighbors, amr_flags_of_neighbors));
+              }
+              *element = Element<volume_dim>(element_id, new_neighbors);
+            },
+            make_not_null(&box));
+      }
+
       // Check for p-refinement
       if (alg::any_of(my_amr_flags, [](amr::Flag flag) {
             return (flag == amr::Flag::IncreaseResolution or
                     flag == amr::Flag::DecreaseResolution);
           })) {
         db::mutate<::domain::Tags::Mesh<volume_dim>>(
-            [&my_amr_flags](const gsl::not_null<Mesh<volume_dim>*> mesh) {
-              *mesh = amr::projectors::mesh(*mesh, my_amr_flags);
+            [&old_mesh,
+             &my_amr_flags](const gsl::not_null<Mesh<volume_dim>*> mesh) {
+              *mesh = amr::projectors::mesh(old_mesh, my_amr_flags);
             },
             make_not_null(&box));
       }
 
-      // Need to reset AMR flags and determine new neighbors
-      db::mutate<::domain::Tags::Element<volume_dim>,
-                 amr::Tags::Flags<volume_dim>,
+      tmpl::for_each<typename Metavariables::amr_mutators>(
```

Comment:
I think it would be helpful to keep/add the comment that currently only the (de)refined elements run the AMR mutators, and their neighbors don't get a chance to update. We may want to allow neighbors to run AMR mutators as well in the future.

Reply:
actually here the mutators are run on all elements (that are not newly created via h-refinement) regardless of whether their Mesh changed.  I'll add a comment here and mention this in the protocol